### PR TITLE
get documentation building

### PR DESCRIPTION
**Pull request by justinkao**
- add sphinx build files for documentation
- copied over some style and template files from the main IPython repo
- modify `_version.py` to contain more version info
- move `examples/Parallel Computing` down into `examples/` and update links
- move `docs/source/parallel` down into `docs/source`
- `make` in the `docs/` directory now generates html output

### CODE REVIEW

File: ipython_parallel/_version.py [hunk 1]
```diff
@@ -1,2 +1,9 @@
-version_info = (4, 0, 0, 'dev')
+_version_major = 4
+_version_minor = 0
+_version_patch = 0
+_version_extra = 'dev'
```
Comment (minrk) [inline ipython_parallel/_version.py]: Can you revert the changes in _version_.py, please and leave it how it was? It was like that for a reason.

Comment (justinkao) [inline ipython_parallel/_version.py]: The conf script for ipython uses `_version_extra` to insert a disclaimer:
https://github.com/ipython/ipython/blob/master/docs/source/conf.py#L72

I wanted to also support this, should I take it out?

Comment (takluyver) [inline ipython_parallel/_version.py]: See what we did for nbconvert - left it basically the same but added _version_extra:

https://github.com/jupyter/jupyter_nbconvert/pull/1/files#diff-d7781741d938b7b8e52a1059d3772880

Comment (minrk) [inline ipython_parallel/_version.py]: I would indeed take out the check for version_extra in conf.py and leave _version.py exactly how it is.

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# The suffix(es) of source filenames.
+# You can specify multiple suffix as a list of string:
+# source_suffix = ['.rst', '.md']
+source_suffix = '.rst'
+
+# Add dev disclaimer.
+if iprelease['_version_extra'] == 'dev':
```
Comment (minrk) [inline docs/source/conf.py]: This can be replaced with `_version.version_info[-1] == 'dev'`. There is no need for any changes in `_version.py`

File: docs/source/conf.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+    """
+
+# The encoding of source files.
+#source_encoding = 'utf-8-sig'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = u'IPython.parallel'
```
Comment (minrk) [inline docs/source/conf.py]: ipython_parallel

File: docs/source/conf.py [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+author = u'The IPython Development Team'
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = iprelease['version'].split('-', 1)[0]
+# The full version, including alpha/beta/rc tags.
+release = iprelease['version']
```
Comment (minrk) [inline docs/source/conf.py]: release and version will always have the same value here.

File: docs/source/parallel_demos.rst [hunk 5]
```diff
@@ -163,11 +163,11 @@ calculation can also be run by simply typing the commands from
      'pi200m.ascii.13of20',
      'pi200m.ascii.14of20',
      'pi200m.ascii.15of20']
-    
+
     # download the data files if they don't already exist:
     In [8]: v.map(fetch_pi_file, files)
-
```
Comment (minrk) [inline docs/source/parallel_demos.rst]: for future reference, please avoid stripping trailing whitespace.

File: docs/source/conf.py (hunk 4 above)
Comment (justinkao) [inline docs/source/conf.py]: Should it come out to be '4.0.0.dev' or '4.0.0'?

Comment (minrk) [inline docs/source/conf.py]: From the sphinx docs, I think `release` should be `__version__`, and `version` should be `'.'.join(map(str, version_info[:2]))`

**Comment by minrk** [comment]
> copied over some style and template files from the main IPython repo

let's not inherit any style or templates from IPython. That stuff is super old.

**Comment by minrk** [comment]
@justinkao are you planning to address comments on this, or should we finish it up?

**Comment by minrk** [comment]
Finished up in #5
